### PR TITLE
Order the fields in dex env

### PR DESCRIPTION
**Pull request by rene-dekker**
It turns out that the idp secret's fields are not always read in the same order. This can cause the envvars in the spec of the dex deployment to change periodically/perpetually, without any actual changes.

### CODE REVIEW

**Review by Suraiya-Hameed — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

File: pkg/render/dex_config.go
Comment (Brian-McM) [inline pkg/render/dex_config.go:336]: Nit, I'm not really a fan of anonymous, sometimes there's a need for them but I'm not sure this is one of those times.

**Review by Brian-McM — APPROVED** [review]

**Pull request merged**
